src/str/transform.php

```diff
@@ -294,6 +294,36 @@ function to_int(
   return null;
 }
 
+/**
+ * Returns the "haystack" string with all occurrences of the keys of
+ * `$replacements` replaced by the corresponding values.
+ *
+ * Once a substring has been replaced, its new value will not be searched
+ * again.
+ *
+ * - For having new values searched again, see `Str\replace_every()`.
+ */
+<<__Rx>>
+function translate(
+  string $haystack,
+  KeyedContainer<string, string> $replacements,
+): string {
+  invariant(
+    /* HH_FIXME[2049] calling stdlib directly */
+    /* HH_FIXME[4107] calling stdlib directly */
+    /* HH_FIXME[4200] Rx calling non-rx */
+    \min(\array_map($key ==> \strlen($key), \array_keys($replacements))) > 0,
```

Comment:
this has has memory and cpu complexity that aren't required.
memory: allocates intermediary arrays
cpu: performance is relative to the number of elements in `$replacements` for a number of operations

there can be exactly one key that matches this condition and it is O(1) to check it in a `KeyedContainer`.
```
invariant(
  !C\contains_key($replacements, ''),
  'Expected non-empty keys only.',
);
```

Reply:
Thanks! Just pushed your suggestion.